app/validators/not_blocked_validator.rb

```diff
@@ -0,0 +1,27 @@
+class NotBlockedValidator < ActiveModel::EachValidator
+  include BlockHelper
+
+  def validate_each(record, attribute, value)
+    return if value.nil?
+
+    blocker = options[:by]
+    case blocker
+    when Symbol
+      blocker = record.send(blocker)
+    when Proc
+      blocker = blocker.call(record)
+    end
+
+    blocker_users = users_for(blocker)
+    blocked_users = users_for(value)
+
+    # You can't be blocked from interacting with your own things:
+    return if (blocked_users - blocker_users).empty?
+
+    return unless Block.exists?(blocker: blocker_users, blocked: blocked_users)
+
+    record.errors.add(attribute, options.fetch(:message, :blocked),
+                      blocked: blocked_users.map(&:login).to_sentence,
```

Comment:
They're currently unused, yes. I added it so that this validator will be usable even in (potential future) cases where the error message is supposed to contain more information than just "That user has blocked you." I can remove them for now if you'd like.